### DISCUSSION TITLE
关于收集课程评价及后续发展的想法

### DISCUSSION THREAD

**Discussion by Crescentm** [openwhu/OpenWHU#60] [Q&A]
## 前言
无论是梦想珈还是淘课啦，都曾打算做一个课程评价的系统，想法是十分美好的，但最终所得到的只有学校的封锁。
窃以为，评价课程看的不应是上课所能得到的绩点，而是包括学生和老师的方法面面的。
所以或许，**只收集关于课程的评论**是一个不错的想法。
## 想法
最近的OpenWHU又回到了以前的状态，但一时的热度确实也不能改变OpenWHU不方便的事实，这样断断续续的Pr从某种角度来说可能才是正常的。（其实最主要问题是我一直在摸鱼）但我并不想满足于此。

OpenWHU的最初的本意是收集课程的资料，但为WHU的学生提供一个自由交流的平台也是我的愿望。
曾与课程交流群交流，想将这些评论放在github上，但收集和查看对于普通学生都是十分困难的。
对于广大的学生而言，在微信小程序或者app上像豆瓣那样评分+评论一个课程才是最想要的。
天下没有免费的午餐，要搭建这样一个便利的平台，程序员、服务器、人脉、时间都会是问题。如何在为爱发电的基础上为尽可能多的学生提供便利的服务估计是每一个热爱这个学校的人都会想到的问题。
## 问
在这写下这些，其实是想听听大家的意见，如何做出这样一个便利的平台，和如何继续发展OpenWHU。
## 后言
深夜写了这些想法，或许最终也只能成为臆想，但万一有机会呢？

**Comment by Crescentm** [comment]
微信小程序说不定是一个好主意，但我不会开发小程序😭

**Comment by yqwu905** [comment]
非常支持，我个人觉得网页端应该还不错，可以和OpenWHU项目整合一下(大概类似于一个课程内可以同时看到评价和相关资料?) 这样的话也许网页端是个不错的选择(微信小程序的话电脑端访问有点麻烦, 而且下载文件似乎不是很方便)
我完全不擅长网页开发, 希望有擅长这方面的同学能来做,但如果需要的话应该能勉强整出来一个能用的版本(不过这个月没时间, 如果大家确实有这个打算的话1月可以开始弄)
资金方面, 我这两年应该一直可以提供服务器的说, 但是资金来源仍然是一个问题, 全靠维护者用爱发电并不是很现实, 靠捐赠又有违这个项目的初衷.

**Comment by Crescentm** [comment]
今天在群里看到了北大的网站，突然有了前进的目标。

https://courses.pinzhixiaoyuan.com/

**Comment by Linorman** [comment]
前端倒是可以用web实现，有余力的情况下可以搞个multi- platform（虽然大概率要开发两套web和移动端分开）😂；我觉得更关键的可能是后台设计，毕竟武大课很多，这个项目又比较依赖合理的文件存储读写方式，也许后端需要花点时间设计。如果有相关计划的话可以推广，招一些程序和策划美工（如果考虑规模做大的情况下），资金也可以众筹，但是整个项目的启动可能得有详细的计划，要不极有可能难产。希望这个项目能搞起来，给武大学生带来便利😭（我虽然菜但也可以提供技术力🫡）。
